onnxruntime/core/optimizer/propagate_cast_ops.cc

```diff
@@ -141,18 +161,29 @@ static bool IsFP16Allow(const std::string& op_type, size_t level, const FP16Allo
 static bool IsCastTo(const Node* node, TensorProto_DataType data_type) {
   if (node->OpType() == "Cast") {
     const NodeAttributes& attributes = node->GetAttributes();
-    ORT_ENFORCE(attributes.find("to") != attributes.end());
+    const auto attr_hit = attributes.find("to");
+    ORT_ENFORCE(attr_hit != attributes.end(), "Node: ", node->Name(),
+                " is a Cast node and it must have 'to' attribute set.");
     const NodeArg* input = node->InputDefs()[0];
     auto input_data_type = static_cast<TensorProto_DataType>(input->TypeAsProto()->tensor_type().elem_type());
     // Allow cast nodes with same input and output type float/float16 to eliminate such casts.
     return (input_data_type == TensorProto::FLOAT16 || input_data_type == TensorProto::FLOAT) &&
-           attributes.at("to").i() == static_cast<int64_t>(data_type);
+           attr_hit->second.i() == static_cast<int64_t>(data_type);
   }
   return false;
 }
 
+static void AddToAttribute(TensorProto_DataType data_type, NodeAttributes& attributes) {
```

Comment:
This function is only called in two places. Is it possible to inline this function? Does adding a function justify the function call overhead?

Reply:
The decision to inline ultimately belongs to the compiler. I would rather prefer not to inline it for the following reasons:
* There is not a profiler evidence that it creates an overhead
* The function as written avoids copying and extra memory allocation as compared to the original code, and it is highly likely to be more efficient.
* Given the fact it calls many inline functions, it would be big enough not to inline.
* it improves code organization, does not increase binary size and provides the correct way to add attribute.

Reply:
Is the profiler evidence that it DOES NOT create overhead?